File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.ts
function findDuplicate(nums: number[]): number {
    let n = nums.length;
    nums.sort((a,b) => a-b);
    for(let i=1;i<n;i++){
        if(nums[i-1]==nums[i]) return nums[i];
    }
    return -1;
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n=nums.size();
        //find the duplicate number
        sort(begin(nums),end(nums));
        for(int i=1;i<size(nums);i++){
            if(nums[i-1] == nums[i]){
                return nums[i];
            }
        }
        return -1;
    }
};